draft-irtf-cfrg-aegis-aead.md

```diff
@@ -1620,6 +1620,201 @@ Instead of relying on the generic `Encrypt` function, implementations can skip t
 
 After initialization, the `Update` function is called with constant parameters, allowing further optimizations.
 
+# AEGIS as a Message Authentication Code
+
+All AEGIS variants can be used to construct a MAC.
+
+For all the variants, The `Mac` function takes as input a key, a nonce, and data, and produces a 128- or 256-bit tag as output.
+
+This is the only function that permits the reuse of `(key, nonce)` pairs with different inputs.
+
+AEGIS-based MAC functions MUST NOT be used as hash functions. If the key is known, inputs causing state collisions can easily be crafted.
+Likewise, unlike hash-based MACs, tags MUST NOT be used for key derivation, as there is no guarantee they are uniformly random.
+
+~~~
+Mac(data, key, nonce)
+~~~
+
+Inputs:
+
+- `data`: the input data to be authenticate (length MUST be less than or equal to `A_MAX`).
+- `key`: the encryption key.
+- `nonce`: the public nonce.
+
+Outputs:
+
+- `tag`: the authentication tag.
+
+## AEGISMAC-128L
+
+AEGISMAC-128L refers to the `Mac` function built upon the building blocks of AEGIS-128L.
+
+Steps:
+
+~~~
+Init(key, nonce)
+data_blocks = Split(ZeroPad(data, 256), 256)
+for di in data_blocks:
+    Absorb(di)
+tag = Finalize(|data|)
+return tag
+~~~
+
+It is equivalent to an evaluation of the `Encrypt` function of AEGIS-128L with the input data as the `ad` and leaving `msg` empty, resulting in just a tag.
+
+## AEGISMAC-256
+
+Similarily, AEGISMAC-256 refers to the `Mac` function build upon the building blocks of AEGIS-256.
+
+Steps:
+
+~~~
+Init(key, nonce)
+data_blocks = Split(ZeroPad(data, 128), 128)
+for di in data_blocks:
+    Absorb(di)
+tag = Finalize(|data|)
+return tag
+~~~
+
+## AEGISMAC-128X
+
+The AEGISMAC-128X MAC is based on the building blocks of AEGIS-128X but replaces the `Finalize` function with a dedicated `FinalizeMac` function.
```

Comment:
> replaces the `Finalize` function with a dedicated `FinalizeMac` function

Does this apply to AEGISMAC-128L and -256 as well? Because their `Finalize()` function takes two lengths: `Finalize(ad_len_bits, msg_len_bits)`.

Reply:
Scrolling down, there's no separate `Finalize()` function for those, so I think AEGISMAC-128L and -256 should have `Finalize(|data|)` changed to account for the other length.

Reply:
Indeed, for 128L and 256, it should be `Finalize(|data|, 0)`. Fixed, thanks!